packages/x-scheduler-headless/src/use-agenda-event-occurrences-grouped-by-day/useAgendaEventOccurrencesGroupedByDay.ts

```diff
@@ -0,0 +1,138 @@
+'use client';
+
+import * as React from 'react';
+import { useStore } from '@base-ui-components/utils/store';
+import { useAdapter, diffIn } from '../use-adapter/useAdapter';
+import { useEventCalendarStoreContext } from '../use-event-calendar-store-context';
+import { selectors } from '../use-event-calendar';
+import { useDayList } from '../use-day-list';
+import { CalendarProcessedDate, CalendarEventOccurrence } from '../models';
+import { innerGetEventOccurrencesGroupedByDay } from '../use-event-occurrences-grouped-by-day';
+import { AGENDA_VIEW_DAYS_AMOUNT, AGENDA_MAX_HORIZON_DAYS } from '../constants';
+
+/**
+ * Agenda-specific hook that:
+ *  - Builds the day list starting at `date`
+ *  - Groups event occurrences by day
+ *  - If `showEmptyDays` is false, extends the range forward until it fills `amount` days that contain events (up to a horizon limit)
+ */
+export function useAgendaEventOccurrencesGroupedByDay(): useAgendaEventOccurrencesGroupedByDayOptions.ReturnValue {
+  const adapter = useAdapter();
+  const store = useEventCalendarStoreContext();
+
+  const getDayList = useDayList();
+
+  const events = useStore(store, selectors.events);
+  const visibleDate = useStore(store, selectors.visibleDate);
+  const showWeekends = useStore(store, selectors.showWeekends);
+  const showEmptyDays = useStore(store, selectors.showEmptyDaysInAgenda);
+  const visibleResources = useStore(store, selectors.visibleResourcesMap);
+
+  const amount = AGENDA_VIEW_DAYS_AMOUNT;
+
+  return React.useMemo(() => {
+    if (process.env.NODE_ENV !== 'production') {
+      if (amount <= 0) {
+        throw new Error(
+          `useAgendaEventOccurrencesGroupedByDay: The 'amount' parameter must be a positive number, but received ${amount}.`,
+        );
+      }
+    }
+
+    // 1) First chunk of days
+    let accumulatedDays = getDayList({
+      date: visibleDate,
+      amount,
+      excludeWeekends: !showWeekends,
+    });
+
+    // Compute occurrences for the current accumulated range
+    let occurrenceMap = innerGetEventOccurrencesGroupedByDay(
+      adapter,
+      accumulatedDays,
+      events,
+      visibleResources,
+    );
+
+    const hasEvents = (d: CalendarProcessedDate) => (occurrenceMap.get(d.key)?.length ?? 0) > 0;
```

Comment:
Maybe we can try to avoid shortcuts? 

```suggestion
    const hasEvents = (day: CalendarProcessedDate) => (occurrenceMap.get(day.key)?.length ?? 0) > 0;
```

Reply:
my bad 🙈